runtime/compiler/control/CompilationThread.cpp

```diff
@@ -3119,7 +3159,11 @@ TR::CompilationInfo::startCompilationThread(int32_t priority, int32_t threadId,
    else
       {
       getCompilationMonitor()->enter();
-      _numCompThreads++;
+      if (_numCompThreads < TR::Options::_numUsableCompilationThreads)
```

Comment:
I don't think it would fail. However, the reason for this `if` is so that we don't increment `_numCompThreads` past `_numUsableCompilationThreads`; we need to run `startCompilationThread` for all allocated threads, but we also need to maintain the distinction between the `_numCompThreads` and `_numAllocatedCompThreads`.